src/encoding/address.ts

```diff
@@ -88,101 +162,19 @@ export function isValidAddress(address: string) {
 }
 
-/**
- * fromMultisigPreImg takes multisig parameters and returns a 32 byte typed array public key,
- * representing an address that identifies the "exact group, version, and public keys" that are required for signing.
- * Hash("MultisigAddr" || version uint8 || threshold uint8 || PK1 || PK2 || ...)
- * Encoding this output yields a human readable address.
- * @param version - multisig version
- * @param threshold - multisig threshold
- * @param pks - array of typed array public keys
- */
-export function fromMultisigPreImg({
-  version,
-  threshold,
-  pks,
-}: Omit<MultisigMetadata, 'addrs'> & {
-  pks: Uint8Array[];
-}) {
-  if (version !== 1 || version > 255 || version < 0) {
-    // ^ a tad redundant, but in case in the future version != 1, still check for uint8
-    throw new Error(INVALID_MSIG_VERSION_ERROR_MSG);
-  }
-  if (
-    threshold === 0 ||
-    pks.length === 0 ||
-    threshold > pks.length ||
-    threshold > 255
-  ) {
-    throw new Error(INVALID_MSIG_THRESHOLD_ERROR_MSG);
-  }
-  const pkLen = ALGORAND_ADDRESS_BYTE_LENGTH - ALGORAND_CHECKSUM_BYTE_LENGTH;
-  if (pkLen !== nacl.PUBLIC_KEY_LENGTH) {
-    throw new Error(UNEXPECTED_PK_LEN_ERROR_MSG);
-  }
-  const merged = new Uint8Array(
-    MULTISIG_PREIMG2ADDR_PREFIX.length + 2 + pkLen * pks.length
-  );
-  merged.set(MULTISIG_PREIMG2ADDR_PREFIX, 0);
-  merged.set([version], MULTISIG_PREIMG2ADDR_PREFIX.length);
-  merged.set([threshold], MULTISIG_PREIMG2ADDR_PREFIX.length + 1);
-  for (let i = 0; i < pks.length; i++) {
-    if (pks[i].length !== pkLen) {
-      throw new Error(INVALID_MSIG_PK_ERROR_MSG);
-    }
-    merged.set(pks[i], MULTISIG_PREIMG2ADDR_PREFIX.length + 2 + i * pkLen);
-  }
-  return new Uint8Array(nacl.genericHash(merged));
-}
-
-/**
- * fromMultisigPreImgAddrs takes multisig parameters and returns a human readable Algorand address.
- * This is equivalent to fromMultisigPreImg, but interfaces with encoded addresses.
- * @param version - multisig version
- * @param threshold - multisig threshold
- * @param addrs - array of encoded addresses
- */
-export function fromMultisigPreImgAddrs({
-  version,
-  threshold,
-  addrs,
-}: {
-  version: number;
-  threshold: number;
-  addrs: string[];
-}) {
-  const pks = addrs.map((addr) => decodeAddress(addr).publicKey);
-  return encodeAddress(fromMultisigPreImg({ version, threshold, pks }));
-}
+const APP_ID_PREFIX = new TextEncoder().encode('appID');
 
 /**
  * Get the escrow address of an application.
  * @param appID - The ID of the application.
  * @returns The address corresponding to that application's escrow account.
  */
-export function getApplicationAddress(appID: number | bigint): string {
+export function getApplicationAddress(appID: number | bigint): Address {
   const toBeSigned = utils.concatArrays(APP_ID_PREFIX, encodeUint64(appID));
   const hash = nacl.genericHash(toBeSigned);
-  return encodeAddress(new Uint8Array(hash));
+  return new Address(Uint8Array.from(hash));
```

Comment:
Maybe that makes no sense, it looks like your Application type is a model for all the Application info.  Is there an AppID type that might have this functionality?  That's where it is in the Go typing.

Reply:
Correct, the `Application` type is a generated model from the REST API.

There is no AppID type in any of the SDKs, to my knowledge. Even the Go SDK has a function like this: https://github.com/algorand/go-algorand-sdk/blob/92422eeda956a0ee9e0cd03f56bac5daad8158ca/crypto/crypto.go#L748

Passing around app IDs as "normal" integer values has been standard for a long time, so while I agree this method is perhaps not as discoverable as it could be, it still feels acceptable to me.